### PR TITLE
[15.0][MIG] partner_delivery_zone: Migration to 15.0

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
@Tecnativa TT36480

### CODE REVIEW

**Comment by cesar-tecnativa** [comment]
@CarlosRoca13 please review

**Review by CarlosRoca13 — APPROVED** [review]
Functional and code review

**Comment by cesar-tecnativa** [comment]
@LudLaf could you review this?

**Comment by LudLaf** [comment]
> 

Please add the following repo description (to delivery-carrier repo):
`This repo comprises modules that add functionalities to the delivery of physical products.`

**Comment by LudLaf** [comment]
Please update CONFIGURATION.rst file:

"""
To configure this module you need to:

#. Go to *Sales > Configuration > Delivery Zone* and create required codes
and zones.
"""

**Comment by LudLaf** [comment]
Please update DESCRIPTION.rst file like so:

"""
This module allows to allocate delivery zones to customers. Information is
shown on sale orders and stock pickings.
It also allows to group and search by delivery zone in partners, sale orders
and stock pickings views.
"""

**Comment by LudLaf** [comment]
Please update ROADMAP.rst file this way:

"""
* In future versions, the system will be capable of setting delivery zones
based on partner's zip or postal codes.
"""

**Comment by LudLaf** [comment]
Please update USAGE.rst file:

"""
To use this module you need to:

#. Go to *Sales > Configuration > Delivery Zone*.
#. Create required delivery zones and codes.
#. Go to *Sales > Orders > Customers*.
#. Choose a customer.
#. Click on 'Edit' button.
#. Go to *Sales & Purchase* tab.
#. In *Delivery Zone* field select an option for this partner.
#. Click on 'Save' button to apply changes.
#. Go to *Sales > Orders > Quotations*.
#. Create a quotation. Delivery zone field is automatically filled. Confirm
   quotation.
#. Open picking or 'Delivery' info (top right). In 'Additional Information' tab
   you can see a delivery zone has been added.
"""

File: partner_delivery_zone/readme/USAGE.rst [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+#. Go to Sales > Configuration > Delivery Zone.
+#. Create required delivery zones and codes.
+#. Go to Sales > Orders > Customers.
+#. Choose a customer.
+#. Click on 'Edit' button.
+#. Go to Sales & Purchase tab.
+#. In Delivery Zone field select an option for this partner.
+#. Click on 'Save' button to apply changes.
+#. Go to Sales > Orders > Quotations.
+#. Create a quotation. Delivery zone field is automatically filled. Confirm quotation.
```
Comment (LudLaf) [inline partner_delivery_zone/readme/USAGE.rst]: Add 'Choose customer', so instruction looks like: =>

"#. Create a quotation. Choose customer. Delivery zone field is automatically filled. Confirm quotation."

**Comment by cesar-tecnativa** [comment]
> Please update USAGE.rst file:
> 
> """ To use this module you need to:
> 
> #. Go to _Sales > Configuration > Delivery Zone_. #. Create required delivery zones and codes. #. Go to _Sales > Orders > Customers_. #. Choose a customer. #. Click on 'Edit' button. #. Go to _Sales & Purchase_ tab. #. In _Delivery Zone_ field select an option for this partner. #. Click on 'Save' button to apply changes. #. Go to _Sales > Orders > Quotations_. #. Create a quotation. Delivery zone field is automatically filled. Confirm quotation. #. Open picking or 'Delivery' info (top right). In 'Additional Information' tab you can see a delivery zone has been added. """

done

**Comment by cesar-tecnativa** [comment]
> Please update USAGE.rst file:
> 
> """ To use this module you need to:
> 
> #. Go to _Sales > Configuration > Delivery Zone_. #. Create required delivery zones and codes. #. Go to _Sales > Orders > Customers_. #. Choose a customer. #. Click on 'Edit' button. #. Go to _Sales & Purchase_ tab. #. In _Delivery Zone_ field select an option for this partner. #. Click on 'Save' button to apply changes. #. Go to _Sales > Orders > Quotations_. #. Create a quotation. Delivery zone field is automatically filled. Confirm quotation. #. Open picking or 'Delivery' info (top right). In 'Additional Information' tab you can see a delivery zone has been added. """

done

**Comment by cesar-tecnativa** [comment]
> Please update USAGE.rst file:
> 
> """ To use this module you need to:
> 
> #. Go to _Sales > Configuration > Delivery Zone_. #. Create required delivery zones and codes. #. Go to _Sales > Orders > Customers_. #. Choose a customer. #. Click on 'Edit' button. #. Go to _Sales & Purchase_ tab. #. In _Delivery Zone_ field select an option for this partner. #. Click on 'Save' button to apply changes. #. Go to _Sales > Orders > Quotations_. #. Create a quotation. Delivery zone field is automatically filled. Confirm quotation. #. Open picking or 'Delivery' info (top right). In 'Additional Information' tab you can see a delivery zone has been added. """

done

**Comment by cesar-tecnativa** [comment]
> > 
> 
> Please add the following repo description (to delivery-carrier repo): `This repo comprises modules that add functionalities to the delivery of physical products.`

done

**Comment by cesar-tecnativa** [comment]
> Please update USAGE.rst file:
> 
> """ To use this module you need to:
> 
> #. Go to _Sales > Configuration > Delivery Zone_. #. Create required delivery zones and codes. #. Go to _Sales > Orders > Customers_. #. Choose a customer. #. Click on 'Edit' button. #. Go to _Sales & Purchase_ tab. #. In _Delivery Zone_ field select an option for this partner. #. Click on 'Save' button to apply changes. #. Go to _Sales > Orders > Quotations_. #. Create a quotation. Delivery zone field is automatically filled. Confirm quotation. #. Open picking or 'Delivery' info (top right). In 'Additional Information' tab you can see a delivery zone has been added. """

done

**Comment by cesar-tecnativa** [comment]
> Please update USAGE.rst file:
> 
> """ To use this module you need to:
> 
> #. Go to _Sales > Configuration > Delivery Zone_. #. Create required delivery zones and codes. #. Go to _Sales > Orders > Customers_. #. Choose a customer. #. Click on 'Edit' button. #. Go to _Sales & Purchase_ tab. #. In _Delivery Zone_ field select an option for this partner. #. Click on 'Save' button to apply changes. #. Go to _Sales > Orders > Quotations_. #. Create a quotation. Delivery zone field is automatically filled. Confirm quotation. #. Open picking or 'Delivery' info (top right). In 'Additional Information' tab you can see a delivery zone has been added. """

done

**Comment by cesar-tecnativa** [comment]
@pedrobaeza could you plesae review?

**Comment by cesar-tecnativa** [comment]
@sergio-teruel please review

File: partner_delivery_zone/__manifest__.py [hunk 2]
```diff
@@ -2,8 +2,9 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 {
     "name": "Partner Delivery Zone",
-    "summary": "Set on partners a zone for delivery goods",
-    "version": "14.0.1.1.0",
+    "summary": "This repo comprises modules that add functionalities"
```
Comment (pedrobaeza) [inline partner_delivery_zone/__manifest__.py]: This description is not correct. This is not a repo, but a module, and the rest doesn't apply to the module goal

File: partner_delivery_zone/readme/CONFIGURATION.rst [hunk 3]
```diff
@@ -1,3 +1,4 @@
 To configure this module you need to:
 
-#. Go to *Sales > Configuration> Delivery Zones* and create any zones
+#. Go to Sales > Configuration > Delivery Zone and create required codes
```
Comment (pedrobaeza) [inline partner_delivery_zone/readme/CONFIGURATION.rst]: I think should go in Contacts > Configuration (and use plural in the menu name).

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: partner_delivery_zone/readme/CONFIGURATION.rst (hunk 3 above)
Comment (LudLaf) [reply partner_delivery_zone/readme/CONFIGURATION.rst]: > I think should go in Contacts > Configuration (and use plural in the menu name).



*Delivery Zone* doesn't show up in *Contacts*, but can be found in *Sales*.

File: partner_delivery_zone/__manifest__.py (hunk 2 above)
Comment (LudLaf) [reply partner_delivery_zone/__manifest__.py]: Sorted.

**Comment by LudLaf** [comment]
Just awaiting approval for requested changes. Everything else looks fine.

File: partner_delivery_zone/readme/CONFIGURATION.rst (hunk 3 above)
Comment (pedrobaeza) [reply partner_delivery_zone/readme/CONFIGURATION.rst]: This is not modified

**Review by LudLaf — APPROVED** [review]

File: partner_delivery_zone/readme/CONFIGURATION.rst (hunk 3 above)
Comment (pedrobaeza) [reply partner_delivery_zone/readme/CONFIGURATION.rst]: But we want the other way around.

Comment (LudLaf) [reply partner_delivery_zone/readme/CONFIGURATION.rst]: Readme CONFIGURATION file changed as per your request.

Comment (pedrobaeza) [reply partner_delivery_zone/readme/CONFIGURATION.rst]: But what about moving the menu to the Contacts app?

Comment (cesar-tecnativa) [reply partner_delivery_zone/readme/CONFIGURATION.rst]: ok I will do it

File: partner_delivery_zone/views/partner_delivery_zone_view.xml [hunk 4]
```diff
@@ -48,8 +48,8 @@
         </field>
     </record>
     <menuitem
-        id="partner_delivery_zone_menu"
-        parent="sale.menu_sale_config"
+        id="contacts_partner_delivery_zone_menu"
```
Comment (pedrobaeza) [inline partner_delivery_zone/views/partner_delivery_zone_view.xml]: no need to change the ID

File: partner_delivery_zone/readme/USAGE.rst [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
-#. Go to *Sales > Customers*.
-#. Go to *Sales and Purchases* tab.
-#. Set a delivery zone for this partner in *Delivery Zone* field.
-#. Got to *Sales > Quotations*
-#. Create a quotation, automatically delivery zone field is filled. Confirm
-   it.
-#. Open the picking and you can see the delivery zone in
-   'Aditional Information' tab.
+#. Go to Sales > Configuration > Delivery Zone.
+#. Create required delivery zones and codes.
```
Comment (pedrobaeza) [inline partner_delivery_zone/readme/USAGE.rst]: These 2 lines should go out.

File: partner_delivery_zone/readme/CONFIGURATION.rst [hunk 6]
```diff
@@ -1,3 +1,3 @@
-To configure this module you need to:
+To configure this module, you need to:
```
Comment (pedrobaeza) [inline partner_delivery_zone/readme/CONFIGURATION.rst:1]: The file should be named `CONFIGURE.rst` for being recognized.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: partner_delivery_zone/readme/CONFIGURATION.rst (hunk 6 above)
Comment (LudLaf) [reply partner_delivery_zone/readme/CONFIGURATION.rst:1]: Changes made, and module just pushed.

**Review by LudLaf — APPROVED** [review]

File: partner_delivery_zone/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Partner Delivery Zone",
+    "summary": "This module allow to create partner delivery zones of physical products",
```
Comment (pedrobaeza) [inline partner_delivery_zone/__manifest__.py]:
```suggestion
    "summary": "This module allows to create partner delivery zones for physical products",
```

File: partner_delivery_zone/views/partner_delivery_zone_view.xml (hunk 4 above)
Comment (pedrobaeza) [reply partner_delivery_zone/views/partner_delivery_zone_view.xml]: This is still pending.

File: partner_delivery_zone/__manifest__.py (hunk 7 above)
Comment (LudLaf) [reply partner_delivery_zone/__manifest__.py]: Both changes made.

**Comment by pedrobaeza** [comment]
@LudLaf you have removed the parent on the menu and that's not what I was saying: I was saying that you don't need to change the ID of the menu, to keep the previous one.

**Comment by LudLaf** [comment]
> @LudLaf you have removed the parent on the menu and that's not what I was saying: I was saying that you don't need to change the ID of the menu, to keep the previous one.

@pedrobaeza I got confused when I saw two lines (51 and 52) in red, and believed both lines had to go. My apologies. Now it's been fiixed.

**Comment by pedrobaeza** [comment]
But now the parent is the old one, and we want to be under `Contacts`.

**Comment by LudLaf** [comment]
> But now the parent is the old one, and we want to be under `Contacts`.

~In file "partner_delivery_zone/views/partner_delivery_zone_view.xml" the `menuitem` corresponding to `parent="sale.menu_sale_config"` has been replaced by `parent="contacts.menu_sale_config"`~

Parent added as: `contacts.res_partner_menu_config`

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**